### ISSUE TITLE
Releases rendered at the same time can interfere with each other's templates

### ISSUE THREAD

**Issue by benweissmann** [helm/helm#4888]
@jgdef-tulip and I noticed some strange behavior when installing multiple Helm charts in parallel: one chart sometimes used the other chart's helpers, if they were named the same.

The issue appears to be with this line in Tiller: https://github.com/helm/helm/blob/master/pkg/engine/engine.go#L125 -- calling `Render` modifies the global state of the template engine. The [engine interface](https://github.com/helm/helm/blob/de5a6a931bfc47c1fa3a72db038002cbf68315ff/pkg/tiller/environment/environment.go#L85) indicates that `Render` should be able to be called concurrently, and it look like Tiller [does use the same engine for all requests](https://github.com/helm/helm/blob/de5a6a931bfc47c1fa3a72db038002cbf68315ff/pkg/tiller/release_server.go#L237).

While we only hit this particular issue because we haven't updated our charts to use the new naming convention of `<chart name>.<release name>`. However, even if we did use this naming convention, it seems like there would still be a race where chart `foo` might not be able to access its own templates (regardless of how they're named) if chart `bar` is rendered concurrently.

Does it seem reasonable to add a lock to the template engine to prevent this?

Helm version:

```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

**Comment by bacongobbler** [comment]
I believe https://github.com/helm/helm/pull/4828 intends to fix this. Would you mind testing that PR and provide @seaneagan with any feedback? That would be much appreciated!

**Comment by bacongobbler** [comment]
also might be a duplicate of #4819

**Comment by benweissmann** [comment]
Yes, that does look like the issue we're having. We'll give it a test in our staging environment and report back.

**Comment by bacongobbler** [comment]
This has likely been addressed via #4828 so I'm going to close this, but please re-open if this is not the case. Thanks
